docs/README-node-utils-APIs-xr.md

```diff
@@ -0,0 +1,277 @@
+
+*Note: This document assumes you have a Nexus VM/device at 192.168.0.1 with user/password admin/admin, and an XR VM/device at 192.168.0.2 with user/password admin/admin and grpc port configured to 57777.*
+
+## <a name="minitest-nexus">Run minitest on Nexus</a>
+
+It's a good idea to make sure all the tests are running on Nexus before making any changes (just to be sure).  From the ./cisco_network_node_utils/tests directory:
+
+```bash
+% ruby test_bgp_neighbor.rb -v -- 192.168.0.1 admin admin
+```
+
+All tests should finish successfully (no failures, errors, or skips).
+
+## <a name="minitest-xr">Run minitest on XR</a>
+
+Now run the same tests against XR, to see what is broken and to get an idea of how much work will need to be done.
+
+```bash
+% ruby test_bgp_neighbor.rb -v -- 192.168.0.2:57777 admin admin
+```
+
+## <a name="fix-errors">Fix errors</a>
+
+We'll now go through the types of differences that will cause errors/failures, and how to fix them.  When you've fixed a particular error, run minitest again on that single test until it is working as you intend, then re-run the same test against Nexus to make sure it still works.
+
+#### <a name="get_command">get_command differences</a>
+
+The base get_command parameter will often be different on XR than Nexus.  Usually, the get_command is common across many attributes of a given feature, and therefore will be defined in the _template section.  In the case of the bgp neighbor feature, the get_command parameter to retrieve information about neighbors is "show running bgp all" on Nexus, while it is "show running router bgp" on XR.  This difference caused most/all of the test_bgp_neighbor.rb tests to error when run against XR, which looked something like this:
+
+```bash
+  1) Error:
+TestRouterBgpNeighbor#test_timers:
+Cisco::CliError: CliError: 'show running bgp all' rejected with message:
+'The command 'show running bgp all' was rejected with error:
+
+>>>show-cmd: show running bgp all <<<
+show running bgp all
+              ^
+% Invalid input detected at '^' marker.'
+```
+
+To correct this problem, the following changes were made in bgp_neighbor.yaml:
+
+```
+  _template:
+-   get_command: "show running router bgp"
++   ios_xr:
++     get_command: "show running router bgp"
++   nexus:
++     get_command: "show running bgp all"
+    context:
+      - "router bgp <asnum>"
+      - "(?)vrf <vrf>"
+      - "(?)neighbor <nbr>"
+```
+
+Here, the "global" get_command parameter was converted into two platform-specific parameters, while the context remained the same.
+
+#### <a name="get_value">get\_value/set\_value differences</a>
+
+Similar to the get_command differences above, the feature might have an attribute that is functionally equivalent on Nexus and XR, but have small syntactic differences.  An example of this for bgp neighbor is the "connected-check" attribute which produced the following error on XR:
```

Comment:
nit: connected_check not connected-check, since we're talking about the attribute rather than the CLI.